src/storage-client/src/sink.rs

```diff
@@ -327,7 +327,19 @@ pub async fn publish_kafka_schemas(
 /// timestamp forward.
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
 pub struct LegacyProgressRecord {
-    pub timestamp: Timestamp,
+    // Double Option to tell apart an omitted field from one set to null explicitly
+    // https://github.com/serde-rs/serde/issues/984
+    #[serde(default, deserialize_with = "deserialize_some")]
+    pub timestamp: Option<Option<Timestamp>>,
+}
+
+// Any value that is present is considered Some value, including null.
+fn deserialize_some<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+where
+    T: Deserialize<'de>,
+    D: Deserializer<'de>,
+{
+    Deserialize::deserialize(deserializer).map(Some)
```

Comment:
this is a bit terrifying but thanks for the link!

Reply:
`dtolnay` wrote it, it's from The Book!